apps/emqx_ft/test/emqx_ft_api_SUITE.erl

```diff
@@ -405,23 +411,47 @@ t_configure(Config) ->
         request_json(
             put,
             uri(["file_transfer"]),
-            #{
-                <<"enable">> => true,
-                <<"storage">> => #{
-                    <<"local">> => #{
-                        <<"exporter">> => #{
-                            <<"s3">> => emqx_utils_maps:deep_put(
-                                [<<"transport_options">>, <<"ssl">>, <<"enable">>],
-                                S3Exporter,
-                                false
-                            )
+            emqx_utils_maps:deep_merge(
+                GetConfigJson,
+                #{
+                    <<"enable">> => true,
+                    <<"storage">> => #{
+                        <<"local">> => #{
+                            <<"exporter">> => #{
+                                <<"s3">> => emqx_utils_maps:deep_put(
+                                    [<<"transport_options">>, <<"ssl">>, <<"enable">>],
+                                    S3Exporter,
+                                    false
+                                )
+                            }
                         }
                     }
                 }
-            },
+            ),
             Config
         )
     ),
+    %% put secret as ******, check the secret is unchanged
```

Comment:
This strikes me as an odd design choice: what if user's secret is indeed `******`? It will just be silently discarded. Obviously, it's not possible to have an AWS secret that looks like `******` but it's sort of a general observation.

Design-wise, what if we respond according to slightly different schema for this field, so a response won't be ready for a blind reuse? I.e. `..., "secret_access_key": {"secret":"******"}, ...`. Though I don't know the use case well enough, so take my words with a grain of salt.

Reply:
Right. But after this change it won't be possible to change secret to `******` if I'm reading this right?

Reply:
Maybe if the API obfuscation were to return `null` / `undefined`, then it would be a clearer signal that the intention is truly to set the password to `******`.  :thinking: